scripts/validate_topbraid_uris.py

```diff
@@ -0,0 +1,157 @@
+#
+# ./validate_topbraid_uris.py --curated-file resource_uri_to_clinical_attribute_mapping_file --topbraid-file topbriad_clinical_attribute_sparql_file
+#
+# Author: Manda Wilson
+
+import optparse
+import os.path
+import sys
+import csv
+import re
+
+"""
+Download the current concept URIs for clinical attributes in Topbraid as a TSV file from https://evn.mskcc.org/edg/tbl/swp?_viewClass=endpoint:HomePage:
+
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+        PREFIX cdd:<http://data.mskcc.org/ontologies/clinical_data_dictionary/>
+        PREFIX skos:<http://www.w3.org/2004/02/skos/core#>
+        SELECT ?column_header ?display_name ?attribute_type ?datatype ?description ?priority
+        WHERE {
+            GRAPH <urn:x-evn-master:clinical_data_dictionary> {
+                ?subject skos:prefLabel ?column_header.
+                ?subject cdd:AttributeType ?attribute_type.
+                ?subject cdd:Datatype ?datatype.
+                ?subject cdd:Description ?description.
+                ?subject cdd:DisplayName ?display_name.
+                ?subject cdd:Priority ?priority.
+            }
+        }
+"""
+
+NUM_FIELDS_IN_CURATED_FILE = 2
+URI_PATTERN_STR = "^C[0-9]{6}$" # e.g. C000000
+URI_PATTERN = re.compile(URI_PATTERN_STR)
+COLUMN_HEADER_PATTERN_STR = "^(\")?[A-Z][A-Z_0-9]{0,99}(\"@en)?$" # e.g. "DELIVERED_DOSE"@en or PLATINUM_OS_MONTHS
+COLUMN_HEADER_PATTERN = re.compile(COLUMN_HEADER_PATTERN_STR)
+MULTIPLE_UNDERSCORES_PATTERN = re.compile("_{2,}")
+errors = []
+warnings = []
+
+def validate_uri(uri, filename):
+    """Adds error to errors array if uri does not match expected pattern"""
+    if not URI_PATTERN.match(uri):
+        errors.append("'%s' does not match expected pattern '%s' in file '%s'" % (uri, URI_PATTERN_STR, filename))
+
+def validate_column_header(str, filename):
+    """Adds error to errors array if str does not match expected pattern"""
+    # validate
+    #   has no whitespace
+    #   must begin with letter
+    #   no lower case letters
+    #   less than or equal to 100 characters
+    #   no characters aside from digits
+    #   upper case letters
+    #   underscore character
+    #   do not have 2 underscores in a row
+    if not COLUMN_HEADER_PATTERN.match(str):
+        errors.append("'%s' does not match expected pattern '%s' in file '%s'" % (str, COLUMN_HEADER_PATTERN_STR, filename))
+    if MULTIPLE_UNDERSCORES_PATTERN.search(str):
+        warnings.append("'%s' contains multiple underscores in file '%s'" % (str, filename))
+
+def read_curated_uris(curated_filename):
+    # validate that file is tab delimited
+    uris = {}
+    with open(curated_filename) as curated_file:
+        for line in curated_file:
+            line = line.strip()
+            if "\t" not in line:
+                errors.append("Line '%s' does not have a '\\t' as a delimiter." % (line))
+            fields = line.split()
```

Comment:
if a line were to have 2 consecutive tabs instead of a single tab between columns, I think this would not distinguish that case. perhaps we should explicitly split on the tab character?  Also wouldn't flanking spaces adjacent to the tab be discarded by split() without distinguishing it?

Reply:
I was splitting on '\t' initially but then I couldn't catch the error about the same URI being mapped to different values, because that URI happened to be in a line with a space delimiter.  I think I will go back to splitting on tabs.  After someone fixes the error they get for the missing tab, the updated PR would catch the conflicting URI error.